core/src/test/java/com/linecorp/armeria/common/stream/StreamMessageDuplicatorTest.java

```diff
@@ -241,6 +245,91 @@ public void closeMulticastStreamFactory() {
         assertThatThrownBy(duplicator::duplicateStream).isInstanceOf(IllegalStateException.class);
     }
 
+    // A test for the internal circular queue in AbstractStreamMessageDuplicator.
+    // Queue expansion behaves differently when odd/even number head wrap-around happens.
+    @Test
+    public void circularQueueOddNumHeadWrapAround() {
+        final InnerCircularQueue queue = new InnerCircularQueue(obj -> 4);
+        add(queue, 0, 10);
+        assertThat(queue.size()).isEqualTo(10);
+        queue.requestRemovalAheadOf(8);
+        assertThat(queue.size()).isEqualTo(10); // removing elements happens when adding a element
+
+        int removedLength = queue.addAndRemoveIfRequested(10);
+        assertThat(removedLength).isEqualTo(8 * 4);
+        assertThat(queue.size()).isEqualTo(3); // 11 - 8 elements
+
+        add(queue, 11, 20);
+        queue.requestRemovalAheadOf(20); // head wrap around happens
+        assertThat(queue.elements.length).isEqualTo(16);
+
+        removedLength = queue.addAndRemoveIfRequested(20);
+        assertThat(removedLength).isEqualTo(12 * 4);
+
+        add(queue, 21, 40);              // queue expansion happens
+        assertThat(queue.elements.length).isEqualTo(32);
+        for (int i = 20; i < 40; i++) {
+            assertThat(i).isEqualTo(queue.get(i));
```

Comment:
Shouldn't this be: `assertThat(queue.get(i)).isEqualTo(i)`? (i.e. actual value comes in `assertThat()`)